packages/rum-core/src/domain/rumEventsCollection/action/rageClickChain.ts

```diff
@@ -0,0 +1,123 @@
+import { monitor, ONE_SECOND, timeStampNow } from '@datadog/browser-core'
+import { FrustrationType } from '../../../rawRumEvent.types'
+import type { Click } from './trackClickActions'
+
+export interface RageClickChain {
+  tryAppend: (click: Click) => boolean
+  stop: () => void
+}
+
+export const MAX_DURATION_BETWEEN_CLICKS = ONE_SECOND
+export const MAX_DISTANCE_BETWEEN_CLICKS = 100
+
+const enum RageClickChainStatus {
+  WaitingForMoreClicks,
+  WaitingForClicksToStop,
+  Finalized,
+}
+
+export function createRageClickChain(firstClick: Click): RageClickChain {
+  const bufferedClicks: Click[] = []
+  let stoppedClicksCount = 0
+  let status = RageClickChainStatus.WaitingForMoreClicks
+  let timeout: number | undefined
+  const rageClick = firstClick.clone()
+
+  function dontAcceptMoreClick() {
+    if (status === RageClickChainStatus.WaitingForMoreClicks) {
+      status = RageClickChainStatus.WaitingForClicksToStop
+      tryFinalize()
+    }
+  }
+
+  function tryFinalize() {
+    if (status === RageClickChainStatus.WaitingForClicksToStop && stoppedClicksCount === bufferedClicks.length) {
+      status = RageClickChainStatus.Finalized
+      finalizeClicks(bufferedClicks, rageClick)
+    }
+  }
+
+  function appendClick(click: Click) {
+    click.onStop(() => {
+      stoppedClicksCount += 1
+      tryFinalize()
+    })
+    bufferedClicks.push(click)
+    timeout = setTimeout(monitor(dontAcceptMoreClick), MAX_DURATION_BETWEEN_CLICKS)
+  }
+
+  appendClick(firstClick)
+  return {
+    tryAppend: (click) => {
+      clearTimeout(timeout)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏What about 
- to rename the `timeout` variable to `maxDurationBetweenClicksTimeout`. 
- moving the  `clearTimeout(timeout)` inside `appendClick` next to the `setTimeout(..)`